src/writer.rs

```diff
@@ -962,4 +1044,49 @@ mod tests {
 
         assert!(FdtWriter::new_with_mem_reserv(&non_overlapping).is_ok());
     }
+
+    #[test]
+    fn test_node_name_valid() {
+        assert!(node_name_valid("abcdef"));
+        assert!(node_name_valid("abcdef@1000"));
+        assert!(node_name_valid("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"));
+        assert!(node_name_valid("azAZ09,._+-"));
+        assert!(node_name_valid("Abcd"));
+
+        assert!(node_name_valid(""));
+
+        // Name missing.
+        assert!(!node_name_valid("@1000"));
+
+        // Name too long.
+        assert!(!node_name_valid("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"));
+        assert!(!node_name_valid("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa@1234"));
+
+        // Name contains invalid characters.
```

Comment:
Can we also add a test case with the invalid character `/`? Having that char in the name will split the node in 2 nodes, so I think we should check for it in unit tests as well.

Reply:
Good point - I added a test for a name containing "/".